### ISSUE TITLE
Question about the delivery service

### ISSUE THREAD

**Issue by yangcen6** [allegro/allegro-api#9936]
For this order c101e920-7694-11ef-a613-47b0e7fb1065，we can not find the delivery service in GET /shipment-management/delivery-services
![img_v2_75919d30-b128-47d9-a5d6-8c6b2350e0cl](https://github.com/user-attachments/assets/d63bfb0d-9699-4ab6-98da-ff569939e4ab)

**Comment by Lukasz-Zurek** [comment]
The buyer did choose a delivery method "Allegro Paczkomaty InPost" - id: ce98c310-7694-11ef-a633-71fa9c439d4a.

You should have an own Inpost agreement. Read more [here](https://help.allegro.com/sell/en/a/how-you-can-adjust-ship-with-allegro-to-your-needs-x5Kzwwl9dHl#how-to-add-your-own-agreement). If you add your own Inpost agreement - then this delivery service will be available in the response, when you will call GET /shipment-management/delivery-services

**Comment by yangcen6** [comment]
Is it include the includes Allegro InPost parcel locker Smart?

**Comment by Lukasz-Zurek** [comment]
@yangcen6 I am not sure If I understood your question but own agreement (with InPost) refers to all of the InPost delivery methods.

**Comment by yangcen6** [comment]
Even if the buyer choose the allegro smart like the pituture?
![image](https://github.com/user-attachments/assets/abf67076-d786-4c72-b046-3d8e0a8cbaaf)

**Comment by Lukasz-Zurek** [comment]
Yes, I have just sent you an article, [here is the link](https://help.allegro.com/sell/en/a/how-you-can-adjust-ship-with-allegro-to-your-needs-x5Kzwwl9dHl#how-to-add-your-own-agreement).

**Comment by yangcen6** [comment]
https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV#how-to-create-a-label-for-shipment   
If we integrate the article's interface with our own API system and retrieve shipping number and labels from Allegro, do we also need to upload the agreements with Inpost?

**Comment by Lukasz-Zurek** [comment]
Yes, as long as you use the Ship with Allegro service (described in the link you have sent) via api. However, if you decide to ship directly through a carrier - there will be no such need. But still you will need to provide InPost with correct email address connected with a given orders.

**Comment by yangcen6** [comment]
How can we  ship directly through Inpost？

**Comment by Lukasz-Zurek** [comment]
You should use [InPost Parcel Manager](https://manager.paczkomaty.pl/zaloguj).

**Comment by yangcen6** [comment]
We have upload the aggrement ,but there is a error code:"CREDENTIALS_ID_NOT_PROVIDED"，"commandId":"b1ec82ec-1497-4b50-98b5-ce9f2fbf77b1" 
![img_v2_0af8eac6-d7d8-4764-9eb2-52fe33c3e89l](https://github.com/user-attachments/assets/6e56be43-12b3-471a-83d5-480b72d5249d)

**Comment by Lukasz-Zurek** [comment]
You did not provide a value in the **"credentialsId"** field, while creating a shipment. You have to provide your own agreement id there.

**Comment by Lukasz-Zurek** [comment]
You should receive it from InPost.

**Comment by Lukasz-Zurek** [comment]
@yangcen6 Sorry, I have just focused on the agreement. The **"credentialsId"** can be retrieved via [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getDeliveryServices), together with **"deliveryMethodId"** 2488f7b7-5d1c-4d65-b85c-4cbcf253fd93.

**Comment by yangcen6** [comment]
There is still a error
`{"path_param":{"commandId":"0469ea0e-e3a6-4c18-8bae-20327d721b81"},"shop_id":"123550430"},result={"header":{"code":200,"desc":"success"},"body":{"commandId":"0469ea0e-e3a6-4c18-8bae-20327d721b81","status":"ERROR","errors":[{"code":"INVALID_TOKEN","message":"ShipXtokenprovidedinWZAaccountsettingsismissingorinvalid","details":"ShipXtokenprovidedinWZAaccountsettingsismissingorinvalid","path":"credentialsId","userMessage":"ShipXtokenprovidedinWZAaccountsettingsismissingorinvalid"}],"traceId":"22a751f34050b9a","httpCode":"200"}}`

Here is our Input
```
{
    "body_param": {
        "input": {
            "deliveryMethodId": "2488f7b7-5d1c-4d65-b85c-4cbcf253fd93",
            "credentialsId": "e3d9a326-4358-4455-b8c1-c5946bb697b0",
            "sender": {
                "name": "PL Swiecko DC",
                "company": "XIAOMI",
                "street": "Poland",
                "streetNumber": "850396639",
                "postalCode": "69-100",
                "city": "Poland",
                "countryCode": "PL",
                "email": "[allegro@xiaomi.com](mailto:allegro@xiaomi.com)",
                "phone": "791 444 014"
            },
            "receiver": {
                "name": "Sapulak Joanna",
                "company": "",
                "street": "Warszawa",
                "streetNumber": "856906331",
                "postalCode": "02-677",
                "city": "Warszawa",
                "countryCode": "PL",
                "email": "[djdrl48ng7+ba0572a2@allegromail.pl](mailto:djdrl48ng7+ba0572a2@allegromail.pl)",
                "phone": "+48602194438",
                "point": "WAW79AP"
            },
            "packages": [
                {
                    "type": "PACKAGE",
                    "length": {
                        "value": 17,
                        "unit": "CENTIMETER"
                    },
                    "width": {
                        "value": 8,
                        "unit": "CENTIMETER"
                    },
                    "height": {
                        "value": 2,
                        "unit": "CENTIMETER"
                    },
                    "weight": {
                        "value": 0.084,
                        "unit": "KILOGRAMS"
                    }
                }
            ],
            "labelFormat": "PDF"
        }
    },
    "shop_id": "123550430"
}
```

**Comment by Lukasz-Zurek** [comment]
`"ShipX token provided in WZA account settings is missing or invalid"` - are you sure you have already added your own agreement (with InPost) In [Order settings](https://salescenter.allegro.com/orders/settings)? 

**[Here is an instruction how to do it](https://help.allegro.com/sell/en/a/how-you-can-adjust-ship-with-allegro-to-your-needs-x5Kzwwl9dHl#how-to-add-your-own-agreement).**

**Comment by yangcen6** [comment]
Yes, we have done it
![{A48F05EA-69E5-461B-A8FF-75DE8D127517}](https://github.com/user-attachments/assets/eb0c7763-94d5-48fa-9ef6-54838a0b6361)

**Comment by Lukasz-Zurek** [comment]
@yangcen6 First of all please verify shipX token directly with InPost, if it is correct and if it is dedicated for this specific delivery method too.

**Comment by yangcen6** [comment]
Does this mean that if the buyer chooses a different delivery method of inpost（like Parcel lockers and InPost courier）, they will need a different token?

**Comment by Lukasz-Zurek** [comment]
Possibly, but I do not know as this case is out of our hands and not related strictly to API. Please ask InPost.
